n8n-io/n8n
# fix(Postgres Node): Special datetime values cause errors

## Summary



The Postgres Node uses the [pg module](https://vitaly-t.github.io/pg-promise/module-pg-promise.html) to parse the postgres query response. However, this would fail for some [special datetime values in postgres](https://www.postgresql.org/docs/current/datatype-datetime.html#DATATYPE-DATETIME-SPECIAL-VALUES), e.g. `infinity`, and throw an error `"Invalid time value"`.

This PR adds a check to catch these and return them unparsed. 

Since the modified parse function is only for `TIMESTAMP` or `TIMESTAMPTZ` values, the only other possible edge case would be `null`, which is returned correctly. The test table covers that.

### To Test: 

1. Set up Postgres Test Table: 
```
CREATE TABLE IF NOT EXISTS special_dates
(
   ts timestamp with time zone DEFAULT now()
);

INSERT INTO special_dates (ts)
VALUES
    ('infinity'::timestamptz),
    ('-infinity'::timestamptz),
    ('epoch'::timestamptz),
    ('now'::timestamptz),
    ('today'::timestamptz),
    ('tomorrow'::timestamptz),
    ('yesterday'::timestamptz)
    (NULL);
```
2. Run the Postgres - Execute Query Node with this query: 
```
SELECT * FROM special_dates;
```
3. All rows should be output and no error. 
4. 
![image](https://github.com/user-attachments/assets/2b50f029-bc19-4922-abd4-ac3479dba83e)


## Related Linear tickets, Github issues, and Community forum posts



- https://linear.app/n8n/issue/NODE-1245/postgres-special-datetime-values-causes-errors
- https://github.com/n8n-io/n8n/issues/8912

## Review / Merge checklist
